src/main/java/gregtech/common/metatileentities/multi/electric/generator/LargeTurbineWorkableHandler.java

```diff
@@ -1,153 +1,150 @@
+    private boolean shouldRun = true;
 
-    public LargeTurbineWorkableHandler(MetaTileEntityLargeTurbine metaTileEntity, FuelRecipeMap recipeMap, Supplier<IEnergyContainer> energyContainer, Supplier<IMultipleTankHandler> fluidTank) {
-        super(metaTileEntity, recipeMap, energyContainer, fluidTank, 0L);
-        this.largeTurbine = metaTileEntity;
+    public LargeTurbineWorkableHandler(RecipeMapMultiblockController metaTileEntity, int tier) {
+        super(metaTileEntity);
+        this.BASE_EU_OUTPUT = (int) GTValues.V[tier] * 2;
     }
 
     @Override
     public void update() {
         super.update();
-        MetaTileEntityRotorHolder rotorHolder = largeTurbine.getAbilities(MultiblockAbility.ABILITY_ROTOR_HOLDER).get(0);
-        if (!rotorHolder.isHasRotor()) {
-            setActive(false);
-        }
-        long totalEnergyOutput = getRecipeOutputVoltage();
-        if (totalEnergyOutput > 0) {
-            energyContainer.get().addEnergy(totalEnergyOutput);
+        //update the status of the workable twice per second to reduce the density of checks
+        if (!metaTileEntity.getWorld().isRemote && metaTileEntity.getOffsetTimer() % 10 == 0) {
+            shouldRun = shouldRunUpdate();
         }
     }
 
-    public FluidStack getFuelStack() {
+    public FluidStack getInputFluidStack() {
         if (previousRecipe == null)
             return null;
-        FluidStack fuelStack = previousRecipe.getRecipeFluid();
-        return fluidTank.get().drain(new FluidStack(fuelStack.getFluid(), Integer.MAX_VALUE), false);
+        FluidStack fuelStack = previousRecipe.getFluidInputs().get(0);
+        return getInputTank().drain(new FluidStack(fuelStack.getFluid(), Integer.MAX_VALUE), false);
     }
 
-    @Override
-    public boolean checkRecipe(FuelRecipe recipe) {
-        MetaTileEntityRotorHolder rotorHolder = largeTurbine.getAbilities(MultiblockAbility.ABILITY_ROTOR_HOLDER).get(0);
-        if (++rotorCycleLength >= CYCLE_LENGTH) {
-            int damageToBeApplied = (int) Math.round(BASE_ROTOR_DAMAGE * rotorHolder.getRelativeRotorSpeed()) + 1;
-            if (rotorHolder.applyDamageToRotor(damageToBeApplied, false)) {
-                this.rotorCycleLength = 0;
-                return true;
-            } else return false;
-        }
-        return true;
+    public boolean shouldRun() {
+        return shouldRun;
     }
 
-    @Override
-    public long getMaxVoltage() {
-        MetaTileEntityRotorHolder rotorHolder = largeTurbine.getAbilities(MultiblockAbility.ABILITY_ROTOR_HOLDER).get(0);
-        if (rotorHolder.hasRotorInInventory()) {
-            double rotorEfficiency = rotorHolder.getRotorEfficiency();
-            double totalEnergyOutput = (BASE_EU_OUTPUT + getBonusForTurbineType(largeTurbine) * rotorEfficiency);
-            return MathHelper.ceil(totalEnergyOutput);
-        }
-        return BASE_EU_OUTPUT + getBonusForTurbineType(largeTurbine);
+    public boolean shouldRunUpdate() {
```

Comment:
Instead of checking this in a new method, should this not just override `canRecipeProgress()`